perfarce.py

```diff
@@ -479,42 +495,42 @@ def maxargs(self):
     def run(self, cmd, files=[], abort=True, client=None):
         'Run a P4 command and yield the objects returned'
 
-        c = ['p4', '-G']
+        c = [b'p4', b'-G']
         if self.server:
-            c.append('-p')
+            c.append(b'-p')
             c.append(self.server)
         if client or self.client:
-            c.append('-c')
+            c.append(b'-c')
             c.append(client or self.client)
         if self.root:
-            c.append('-d')
+            c.append(b'-d')
             c.append(shellquote(self.root))
 
         if files and len(files)>self.maxargs:
             tmp = TempFile('w')
             for f in files:
-                if self.ui.debugflag: self.ui.debug('> -x %s\n' % f)
-                print >>tmp.File,f
+                if self.ui.debugflag: self.ui.debug(b'> -x %s\n' % f)
+                print(f, file=tmp.File)
             tmp.close()
-            c.append('-x')
+            c.append(b'-x')
             c.append(tmp.Name)
             files = []
 
         c.append(cmd)
 
-        cs = ' '.join(c + [shellquote(f) for f in files])
-        if self.ui.debugflag: self.ui.debug('> %s\n' % cs)
+        cs = b' '.join(c + [shellquote(f) for f in files])
+        if self.ui.debugflag: self.ui.debug(b'> %s\n' % cs)
 
-        for d in loaditer(os.popen(cs, 'rb')):
-            if self.ui.debugflag: self.ui.debug('< %r\n' % d)
-            code = d.get('code')
-            data = d.get('data')
+        for d in loaditer(popen(cs, b'rb')):
```

Comment:
loaditer expects a file argument. os.popen returns it, but mercurial.utils.procutil.popen does not